modules/apps/accessibility/accessibility-menu-web/src/main/resources/META-INF/resources/js/AccessibilityMenu.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayModal, {useModal} from '@clayui/modal';
+import PropTypes from 'prop-types';
+import React, {useEffect} from 'react';
+
+import AccessibilitySetting from './AccessibilitySetting';
+import {getSettingValue} from './util';
+
+const OPEN_ACCESSIBILITY_MENU_EVENT_NAME = 'openAccessibilityMenu';
+
+const AccessibilityMenu = ({accessibilitySettings}) => {
+	console.log('accessibilitySettings', accessibilitySettings);
+	const {observer, onOpenChange, open} = useModal();
+
+	useEffect(() => {
+		Liferay.on(OPEN_ACCESSIBILITY_MENU_EVENT_NAME, () =>
+			onOpenChange(true)
+		);
+
+		accessibilitySettings.forEach((setting) => {
+			document
+				.querySelector('body')
+				.classList.toggle(
+					setting.className,
+					getSettingValue(
+						setting.defaultValue,
+						setting.sessionClicksValue,
+						setting.key
+					)
+				);
+		});
+
+		return () => {
+			Liferay.detach(OPEN_ACCESSIBILITY_MENU_EVENT_NAME, onOpenChange);
```

Comment:
I could be wrong but I think `Liferay.detach` will just remove the listener based on the function reference too, as above you create an anonymous function it's a different reference to `onOpenChange` for security I think the best would be to create a named function inside the `useEffect` and pass it to the `Liferay.on` and `Liferay.detach` method.

Reply:
Thanks @matuzalemsteles! That's what I meant to do. Good catch! 😄